plugins/embed-optimizer/hooks.php

```diff
@@ -190,7 +190,7 @@ function embed_optimizer_update_markup( WP_HTML_Tag_Processor $html_processor, b
 
 	// As of 1.0.0-beta3, next_tag() allows $query and is beginning to migrate to skip tag closers by default.
 	// In versions prior to this, the method always visited closers and passing a $query actually threw an exception.
-	$tag_query = version_compare( OPTIMIZATION_DETECTIVE_VERSION, '1.0.0-beta3', '>=' )
+	$tag_query = ! defined( 'OPTIMIZATION_DETECTIVE_VERSION' ) || version_compare( OPTIMIZATION_DETECTIVE_VERSION, '1.0.0-beta3', '>=' )
```

Comment:
Why is this function's logic running even when Optimization Detective is not active? Is it not something that relies on that plugin?

Also, looking at this again: Why does the Optimization Detective matter for this, given it's passed to the `WP_HTML_Tag_Processor` instance, which should depend on Core, not OD? Maybe I'm missing something here.

Reply:
Embed Optimizer does not require Optimization Detective. Without Optimization Detective, it lazy-loads _all_ embeds.

When OD is not active, a `WP_HTML_Tag_Processor` instance is passed. When OD is active, then an `OD_HTML_Tag_Processor` instance is passed (a subclass of `WP_HTML_Tag_Processor`).

Reply:
Makes sense, thanks.

I didn't know Embed Optimizer was doing this already on its own (albeit less reliably of course).

Reply:
I'm just realizing that this change and https://github.com/WordPress/performance/pull/1872 probably fix Embed Optimizer to work properly when OD is not active, actually. Because the default behavior of `next_tag()` is to not visit tag closers, and yet the OD subclass was visiting tag closers by default... since no query was being passed in, when OD is not active then it would be skipping all tag closers and the logic that checks `$processor->is_tag_closer()` would always return `false`.

Reply:
Nevermind. The only part of the code that uses `is_tag_closer()` is for the non-OD context:

https://github.com/WordPress/performance/blob/20fea77e8e8e421b42c3b784c4f6fd4136ccb7ed/plugins/embed-optimizer/hooks.php#L212-L234

So there wouldn't have been a bug.